packages/discord.js/src/structures/ApplicationCommand.js

```diff
@@ -325,11 +325,8 @@ class ApplicationCommand extends Base {
     }
 
     if (existing.channelTypes) {
-      const newTypes = (option.channelTypes ?? option.channel_types).map(type =>
-        typeof type === 'number' ? ChannelType[type] : type,
-      );
       for (const type of existing.channelTypes) {
-        if (!newTypes.includes(type)) return false;
+        if (!option.channelTypes.includes(type)) return false;
```

Comment:
This doesn't handle builder data anymore because of the `option.channelTypes ?? option.channel_types` condition missing